### PR TITLE
Added tests for math function migrations

### PR DESCRIPTION
**Pull request by 0x12CC**
This PR adds tests for the following math API functions:

- `__hadd2`
- `__habs`
- `__habs2`
- `labs`
- `llabs`

### CODE REVIEW

File: features/feature_case/math/math-direct.cu [hunk 1]
```diff
@@ -477,6 +480,16 @@ __global__ void math_direct2() {
   __hadd(u, u);
   __mul24(u, u);
   __mulhi(u, u);
+
+  // CHECK: l = sycl::abs(l);
+  // CHECK-NEXT: ll = sycl::abs(ll);
+  l = labs(l);
```
Comment (ziranzha) [inline features/feature_case/math/math-direct.cu]: It is better to check whether result is correct.

File: features/feature_case/math/math-direct.cu [hunk 2]
```diff
@@ -477,6 +480,16 @@ __global__ void math_direct2() {
...
+  ll = llabs(ll);
```
Comment (ziranzha) [inline features/feature_case/math/math-direct.cu]: same as above.

File: features/feature_case/math/math-direct.cu [hunk 3]
```diff
@@ -477,6 +480,16 @@ __global__ void math_direct2() {
...
+
+  // CHECK: h = sycl::fabs(h);
+  // CHECK-NEXT: h2 = sycl::fabs(h2);
+  h = __habs(h);
```
Comment (ziranzha) [inline features/feature_case/math/math-direct.cu]: same as above.

File: features/feature_case/math/math-direct.cu [hunk 4]
```diff
@@ -477,6 +480,16 @@ __global__ void math_direct2() {
...
+  h2 = __habs2(h2);
```
Comment (ziranzha) [inline features/feature_case/math/math-direct.cu]: same as above.

File: features/feature_case/math/math-emu.cu [hunk 5]
```diff
@@ -181,6 +183,8 @@ __global__ void math_emu() {
   __umulhi(u, u2);
   // CHECK: sycl::rhadd(u, u2);
   __urhadd(u, u2);
+  // CHECK: h2 = h2 + h2
+  h2 = __hadd2(h2, h2);
```
Comment (ziranzha) [inline features/feature_case/math/math-emu.cu]: same as above.

**Review by ziranzha — COMMENTED** [review]
hcrp is tested in this PR or in another PR?

**Comment by 0x12CC** [comment]
I've updated the tests to assert the results and added a case for `hrcp`. The `__habs` and `__habs2` functions are in a separate file since they're only supported on GPU devices.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Review by ShengchenJ — APPROVED** [review]
LGTM

**Comment by ShengchenJ** [comment]
@0x12CC Please resolve the conflict. Thanks.

**Comment by 0x12CC** [comment]
> @0x12CC Please resolve the conflict. Thanks.

This should be fixed now.

**Pull request merged**